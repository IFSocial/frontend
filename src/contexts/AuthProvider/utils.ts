import api from '../../services/api';
import { IUser } from './types';

// Salva o usuario no BD local
export function setUserLocalStorage(user: IUser | null) {
  localStorage.setItem('u', JSON.stringify(user));
}

// Checa se existe o usuario no BD local
export function getUserLocalStorage() {
  const json = localStorage.getItem('u');
  if (!json) {
    return null;
  }

  const user = JSON.parse(json);

  return user ?? null;
}

// Faz request do login passando email e senha como payload
export async function LoginRequest(email: string, password: string) {
  try {
    const request = await api.post('/users/signin', { email, password });

    return request.data;
  } catch {
    return null;
  }
}

// Faz request do cadastrp passando nome, email e senha como payload
export async function SignupRequest(
  name: string,
  registration: string,
  sexo: string,
  email: string,
  password: string,
) {
  try {
    const request = await api.post('/users/signup', {
      name,
      registration,
      sexo,
      email,
      password,
    });

    return request.data;
  } catch {
    return null;
  }
}
