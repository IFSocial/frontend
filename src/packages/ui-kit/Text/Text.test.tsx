import React from 'react';

import { Text } from '.';
import { render } from '../../test/utils';

describe('Text', () => {
  it('should render correctly', () => {
    const { getByText } = render(<Text>Text</Text>);

    const textElement = getByText('Text');

    expect(textElement).toBeInTheDocument();
    expect(textElement).toHaveTextContent(/^Text$/);

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text 1', () => {
    const { getByText } = render(<Text size={1}>Text</Text>);

    const textElement = getByText('Text');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text 2', () => {
    const { getByText } = render(<Text size={2}>Text</Text>);

    const textElement = getByText('Text');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text 3', () => {
    const { getByText } = render(<Text size={3}>Text</Text>);

    const textElement = getByText('Text');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });
});
