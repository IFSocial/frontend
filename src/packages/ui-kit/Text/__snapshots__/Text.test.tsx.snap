// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render a text 1 1`] = `
<p
  class="sc-bcXHqe dOBKaS"
  size="1"
>
  Text
</p>
`;

exports[`Text should render a text 2 1`] = `
<p
  class="sc-bcXHqe hCXeSq"
  size="2"
>
  Text
</p>
`;

exports[`Text should render a text 3 1`] = `
<p
  class="sc-bcXHqe eIJVca"
  size="3"
>
  Text
</p>
`;

exports[`Text should render correctly 1`] = `
<p
  class="sc-bcXHqe dOBKaS"
  size="1"
>
  Text
</p>
`;
